neuralprophet/forecaster.py

```diff
@@ -1711,8 +1735,412 @@ def plot_latest_forecast(
                 line_per_origin=True,
             )
 
+    def check_if_configured(self, components, error_flag=False):
+        """Check if components were set in the model configuration by the user.
+
+        Parameters
+        ----------
+            components : str or list, optional
+                name or list of names of components to check
+
+                Options
+                ----
+                * ``trend``
+                * ``trend_rate_change``
+                * ``seasonality``
+                * ``autoregression``
+                * ``lagged_regressors```
+                * ``events``
+                * ``future_regressors`
+                * ``residuals``
+                * ``uncertainty``
+            error_flag : bool
+                Activate to raise a ValueError if component has not been configured
+
+        Returns
+        -------
+            components
+                list of components only including the components set in the model configuration
+        """
+        # Error if selected plotting components are not set in the model configuration by the user
+        for comp in reversed(components):
+            if (
+                ("trend_rate_change" in comp and self.model.config_trend.changepoints is None)
+                or ("seasonality" in comp and self.config_season is None)
+                or ("autoregression" in comp and not self.n_lags > 0)
```

Comment:
> Was a way bigger refactor than I intended, yet happy to see we have less code now overall.
> 
> The original additional `s` in the website workflow is still there. Also added a few other comments, which are rather of cosmetic nature - if the website workflow works again I'm happy - many thanks! :)

Hey @noxan thanks for taking the time to review again!! :) Yes, was also a little surprised that it turned out to be a larger refactor, but happy to have reduced some code lines. sorry about the s in the website workflow! Will keep an eye on it.

Reply:
@noxan regarding the `n_lags`: in forecaster.py in line 387 `self.n_lags = self.config_ar.n_lags`. I assumed it makes no difference. Happy to change it though, if you explain the intention behind it.

Reply:
Just thought it would be more consistent to also check `self.config_ar` (we checked the config for every other component, so why not also for this one)